dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/SdkFeature.kt

```diff
@@ -169,6 +164,27 @@ internal class SdkFeature(
 
     // region Internal
 
+    private fun setupMetricsDispatcher(
+        dataUploadConfiguration: DataUploadConfiguration,
+        filePersistenceConfig: FilePersistenceConfig,
+        context: Context
+    ) {
+        BatchMetricsDispatcher(
+            wrappedFeature.name,
+            dataUploadConfiguration,
+            filePersistenceConfig,
+            internalLogger,
+            coreFeature.timeProvider
+        ).apply {
+            if (context is Application) {
+                context.registerActivityLifecycleCallbacks(
+                    ProcessLifecycleMonitor(this)
+                )
+            }
+            metricsDispatcher = this
+        }
```

Comment:
`apply` already returns the instance.

```suggestion
        metricsDispatcher = BatchMetricsDispatcher(
            wrappedFeature.name,
            dataUploadConfiguration,
            filePersistenceConfig,
            internalLogger,
            coreFeature.timeProvider
        ).apply {
            if (context is Application) {
                context.registerActivityLifecycleCallbacks(
                    ProcessLifecycleMonitor(this)
                )
            }
        }
```